PaddleNLP/language_model/train.py

```diff
@@ -94,221 +91,343 @@ def train():
+                dtype='float32',
+                persistable=True)
+
+            optimizer = fluid.optimizer.SGD(learning_rate=learning_rate)
+            optimizer.minimize(loss)
+
+    # define inference program
+    inference_program = fluid.Program()
+    inference_startup_program = fluid.Program()
+    with fluid.program_guard(inference_program, inference_startup_program):
+        with fluid.unique_name.guard():
+            lm_model.lm_model(
+                config.hidden_size,
+                config.vocab_size,
+                config.batch_size,
+                num_layers=config.num_layers,
+                num_steps=config.num_steps,
+                init_scale=config.init_scale,
+                dropout=config.dropout,
+                rnn_model=config.rnn_model,
+                use_py_reader=False)
+    # Some op behaves differently for train and inference, we need to call
+    # this clone function to ensure every op is right for inference.
+    inference_program = inference_program.clone(for_test=True)
 
     place = fluid.CUDAPlace(0) if args.use_gpu else fluid.CPUPlace()
     exe = Executor(place)
-    exe.run(framework.default_startup_program())
+    exe.run(startup_program)
+
+    device_count = fluid.core.get_cuda_device_count()
+
+    exec_strategy = fluid.ExecutionStrategy()
+    exec_strategy.num_threads = device_count
+    exec_strategy.use_experimental_executor = False
+    exec_strategy.num_iteration_per_drop_scope = 100
+
+    build_strategy = fluid.BuildStrategy()
+    build_strategy.enable_inplace = True
+    build_strategy.memory_optimize = False
+    build_strategy.fuse_all_optimizer_ops = True
+
+    if args.parallel:
+        train_program = fluid.compiler.CompiledProgram(
+            main_program).with_data_parallel(
+                loss_name=loss.name,
+                build_strategy=build_strategy,
+                exec_strategy=exec_strategy)
+    else:
+        train_program = fluid.compiler.CompiledProgram(main_program)
 
     data_path = args.data_path
     print("begin to load data")
     raw_data = reader.ptb_raw_data(data_path)
     print("finished load data")
     train_data, valid_data, test_data, _ = raw_data
 
-    def prepare_input(batch, init_hidden, init_cell, epoch_id=0, with_lr=True):
+    def generate_init_data():
+        init_hidden = np.zeros(
+            (config.num_layers, config.batch_size, config.hidden_size),
+            dtype='float32')
+        init_cell = np.zeros(
+            (config.num_layers, config.batch_size, config.hidden_size),
+            dtype='float32')
+        return init_hidden, init_cell
+
+    def generate_new_lr(epoch_id=0, device_count=1):
+        new_lr = config.base_learning_rate * (config.lr_decay**max(
+            epoch_id + 1 - config.epoch_start_decay, 0.0))
+        if device_count > 1 and args.parallel:
+            lr = np.ones((device_count), dtype='float32') * new_lr
+        else:
+            lr = np.ones((1), dtype='float32') * new_lr
```

Comment:
```
        if device_count > 1 and args.parallel:
            lr = np.ones((device_count), dtype='float32') * new_lr
        else:
            lr = np.ones((1), dtype='float32') * new_lr
```
=> 
```
lr = np.ones((device_count), dtype='float32') * new_lr
```

Reply:
Done. And also remove the setting of `use_experimental_executor`, using the default executor instead.